extensions/data-workspace/src/services/workspaceService.ts

```diff
@@ -24,9 +24,16 @@ export class WorkspaceService implements IWorkspaceService {
 	readonly onDidWorkspaceProjectsChange: vscode.Event<void> = this._onDidWorkspaceProjectsChange?.event;
 
 	private openedProjects: vscode.Uri[] | undefined = undefined;
+	private excludedProjects: string[] | undefined;
 
 	constructor() {
 		this.getProjectsInWorkspace(undefined, true).catch(err => console.error('Error initializing projects in workspace ', err));
+
+		TelemetryReporter.createActionEvent(TelemetryViews.WorkspaceTreePane, TelemetryActions.ProjectsLoaded)
```

Comment:
This is calculating how many projects WS is opened with - this might not be the final number WS ends up with - right?

Reply:
Yup that's right, so this is more for tracking the usage of people using the same workspace and seeing if people actually use a workspace. We can also try matching the machine id from this event with new and open project events in the future to see if there are any patterns there.

Reply:
Hm, what if you sent an event every time you detected a new project added to the workspace? So when first opened up you'd send an event for any in the workspace being opened up but also then send a new event whenever they add a new project to the workspace? That seems more generally useful.

Reply:
there's already a "ProjectAddedToWorkspace" event that gets sent whenever a project gets added. Is that what you had in mind or something different?